test/unstoppable/README.md

```diff
@@ -1,365 +1,308 @@
 And test_unstoppable():
 ```javascript
-/**
-* CODE YOUR SOLUTION HERE
-*/
-function  test_unstoppable() public  checkSolvedByPlayer {
-	Attacker attacker =  new  Attacker();
-	token.transfer(address(attacker), INITIAL_PLAYER_TOKEN_BALANCE);
-	attacker.add_known_address(address(token));
-	attacker.add_known_address(address(vault));
-	attacker.add_known_address(address(monitorContract));
-	attacker.attack();
-}
+function test_unstoppable() public checkSolvedByPlayer {
+        Attacker attacker = new Attacker();
+        token.transfer(address(attacker), INITIAL_PLAYER_TOKEN_BALANCE);
+        attacker.attack();
+    }
 ```
 Let's execute:
-```bash
+```javascript
 forge test -vv --mp test/unstoppable/Unstoppable.t.sol
 ...
 Ran 2 tests for test/unstoppable/Unstoppable.t.sol:UnstoppableChallenge
 [PASS] test_assertInitialState() (gas: 57390)
-[PASS] test_unstoppable() (gas: 332461)
-Suite result: ok. 2 passed; 0 failed; 0 skipped; finished in 2.62ms (331.63µs CPU time)
-
-Ran 1 test suite in 7.44ms (2.62ms CPU time): 2 tests passed, 0 failed, 0 skipped (2 total tests)
+[PASS] test_unstoppable() (gas: 899808)
+Suite result: ok. 2 passed; 0 failed; 0 skipped; finished in 1.56ms 
+Ran 1 test suite in 37.58ms (1.56ms CPU time): 2 tests passed, 0 failed, 0 skipped (2 total tests) 
 ```
 Success! The challenge "Unstoppable" is solved successfully using Halmos symbolic testing.
+
+## And what about fuzzing?
+At the time of writing, [Echidna-driven](https://github.com/crytic/damn-vulnerable-defi-echidna/blob/solutions/contracts/unstoppable/UnstoppableEchidna.sol) solution by Crytic team and [Foundry-driven](https://github.com/devdacian/solidity-fuzzing-comparison/blob/main/test/02-unstoppable/UnstoppableBasicFoundry.t.sol) solution by devdacian can be found on the Internet. However, these solutions were made for older versions of Unstoppable. The current version needs to check other invariants, so we will work with a modernized solution. Of course, I chose invariant-driven Foundry as my fuzzing engine since the current version of Damn-Vulnerable-Defi was completely rewritten on Foundry. 
+### Invariant
+The invariant is obvious - I just took the code of the **_isSolved()** function from **Unstoppable_Halmos.t.sol** and used it:
+```javascript
+function invariant_check_flash_loan() public {
+    vm.prank(deployer);
+    monitorContract.checkFlashLoan(100e18);
+    // Foundry should generate counterexample here.
+    // We expect it to find such a contract and data that sets the vault paused.
+    assert(vault.paused() == false || vault.owner() != deployer);
+}
+```
+### SetUp()
+And the **SetUp()** was changed somewhat, where it was indicated that the attacking actor is only the player, and fuzzing is performed on all known contracts:
+```javascript
+function setUp() public {
+...
+    vm.stopPrank();
+    targetSender(player);
+    targetContract(address(token));
+    targetContract(address(vault));
+    targetContract(address(monitorContract));
```

Comment:
are these necessary? what happens if you don't suggest the target contracts?

Reply:
This is not necessary, but I decided to put it to make it more obvious. Maybe, I should delete it